packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1359,10 +1359,20 @@ export class DeltaManager
         }
 
         // Watch the minimum sequence number and be ready to update as needed
-        assert(this.minSequenceNumber <= message.minimumSequenceNumber, "msn moves backwards");
+        if (this.minSequenceNumber > message.minimumSequenceNumber) {
+            throw new DataCorruptionError("msn moves backwards", {
+                ...extractLogSafeMessageProperties(message),
+                ...(this.connection && { clientId: this.connection?.clientId}),
```

Comment:
```suggestion
                clientId: this.connection?.clientId,
```
minor simplification

Reply:
If there is no concern about `undefined` "pollution"... I'll probably reimplement [`extractLogSafeMessageProperties`](https://github.com/microsoft/FluidFramework/pull/5188/files#diff-01eeb7567dd98442afb9a308001150972f600ef3f04e17a71f88b34a0b8ee153R1462) in a simpler fashion as well

Reply:
i'm not particularly concerned as these objects are short lived, and only created in error cases. if this was on some critical code path we might need to worry